tests/readme.md

```diff
@@ -1,26 +1,41 @@
 
-Definitions of different types of tests can be somewhat subjective. These guidelines are intended to enable `pykale` to have a high level of test coverage, for these tests to address sufficiently compact pieces of code such that it is easy to identify causes of failure and for tests to also cover larger workflows such that confidence can be built in reproducibility of outputs.
+These guidelines will help write tests to address sufficiently compact pieces of code such that it is easy to identify causes of failure and for tests to also cover larger workflows such that confidence or trust can be built in reproducibility of outputs. We use [pytest](https://docs.pytest.org/en/stable/) (see tutorials [python testing software carpentry (alpha)](https://carpentries-incubator.github.io/python-testing/) and [tutorialspoint pytest tutorial](https://www.tutorialspoint.com/pytest/pytest_tutorial.pdf)). There is some subjectivity involved in deciding how much of the potential behaviour of your code to check.
 
-Please use [pykale discussions on testing](https://github.com/pykale/pykale/discussions/categories/testing) to talk about tests and ask for help.
+## Quick start
 
-Refer to the [official pytest documentation](https://docs.pytest.org/en/stable/), or less formal [python testing software carpentry (alpha)](https://carpentries-incubator.github.io/python-testing/) and [tutorialspoint pytest tutorial](https://www.tutorialspoint.com/pytest/pytest_tutorial.pdf), if needed. This will help you to write tests and help with decisions on what aspects of your code need to be tested. There is some subjectivity involved in deciding how much of the potential behaviour of your code to check.
+- [Compact pytest tutorial](https://www.tutorialspoint.com/pytest/pytest_tutorial.pdf)
+- Example [unit tests for deep learning code of Variational Autoencoder (VAE) in PyTorch](https://github.com/tilman151/unittest_dl) and the associated post [How to Trust Your Deep Learning Code](https://krokotsch.eu/cleancode/2020/08/11/Unit-Tests-for-Deep-Learning.html). *Note: it uses `unittest` but we use `pytest`*.
+- Pytest+pytorch examples: [fastai1 tests](https://github.com/fastai/fastai1/tree/master/tests) and [Kornia tests](https://github.com/kornia/kornia/tree/master/test)
+- [Pytest fixtures](https://docs.pytest.org/en/stable/fixture.html), [Pytest exceptions](https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions)
+- [Build-in torch testing modules](https://github.com/pytorch/pytorch/tree/master/torch/testing): `assert_tensors_equal` and `assert_tensors_allclose`
+- [fastai testing](https://fastai1.fast.ai/dev/test.html) is a good high-level reference. Its recommendations on [writing tests](https://fastai1.fast.ai/dev/test.html#writing-tests) are good to follow:
```

Comment:
Some really good stuff in here on how to execute `pytest` in different ways that may be useful in the future. I will learn more.